addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -2142,9 +2147,13 @@ var SnippetsMenu = Widget.extend({
     callPostSnippetDrop: async function ($target) {
         // First call the onBuilt of all options of each item in the snippet
         // (and so build their editor instance first).
-        await this._callForEachChildSnippet($target, function (editor, $snippet) {
-            return editor.buildSnippet();
+        this.postSnippetDropPromise = new Promise(async resolve => {
+            await this._callForEachChildSnippet($target, function (editor, $snippet) {
+                return editor.buildSnippet();
+            });
+            resolve();
         });
+        await this.postSnippetDropPromise;
```

Comment:
@xO-Tx Mmh, why not. If it works to solve the race condition problem, it's good :+1: (as long as we can get rid of this mutex nightmare in master with OWL at some point).

Although:
- Maybe make it so the whole `callPostSnippetDrop` resolution is part of this new Promise? (so create the promise at the start of the method, resolve it at the end?). Would also allow not to indent the existing code.
- await that promise at the start of `cleanForSave` instead of at the end?

Reply:
> * Maybe make it so the whole `callPostSnippetDrop` resolution is part of this new Promise? (so create the promise at the start of the method, resolve it at the end?). Would also allow not to indent the existing code.
> * await that promise at the start of `cleanForSave` instead of at the end?

@qsm-odoo code updated and squashed in the main commit, just let me know if something needs to be changed :+1: